src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -35,8 +35,9 @@
             }
             else
             {
-                Debug.Assert(list.LengthProperty is not null);
-                var lengthEvaluation = new BoundDagPropertyEvaluation(syntax, list.LengthProperty, isLengthOrCount: true, input);
+                Debug.Assert(list.LengthAccess is not null);
+                var lengthProperty = Binder.GetPropertySymbol(list.LengthAccess, out _, out _);
```

Comment:
>lengthProperty [](http://example.com/codeflow?start=20&length=14)

Is this never null?

Reply:
It's only null if there's some error (handled above on line 26 by `list.HasErrors` case). Will add an assertion.